.github/workflows/dev.yml

```diff
@@ -108,52 +119,144 @@ jobs:
+          mv ./target/nextest/default/junit.xml ./target/nextest/default/release.junit.xml
+          jq \
+            --raw-output \
+            --slurp '.[] | select(.type == "test" and (.name | test(".*#\\d+"))) | ( .name | split("#") ) | 
+                     [.[0], (.[1] | tonumber)] | @csv
+          ' ./target/nextest/release.json > ./target/nextest/release.flakes.csv
+          if [ -s ./target/nextest/release.flakes.csv ]; then
+            cargo binstall --no-confirm csview
+            { 
+              echo "FLAKY_TESTS_RELEASE<<EOF"
+              echo -e "\n### :warning: Flaky tests (release run, ${{ matrix.rust.version }})\n";
+              echo "| test | retries |"
+              echo "|------|---------|"
+              csview --style=markdown --no-headers --body-align=left ./target/nextest/release.flakes.csv;
+              echo "EOF"
+            } >> "${GITHUB_ENV}"
+          fi
+
+      - uses: "marocchino/sticky-pull-request-comment@v2"
+        with:
+          header: "flakes_${{matrix.rust.version}}"
+          ignore_empty: 'true'
+          message: |
+            ${{ env.FLAKY_TESTS_DEBUG }}${{ env.FLAKY_TESTS_RELEASE }}
+
+      - name: "publish dev/release test report"
+        uses: "mikepenz/action-junit-report@v5"
+        if: "${{ always() }}"
+        with:
+          annotate_notice: 'false'
+          annotate_only: 'false'
+          check_annotations: 'true'
+          check_retries: 'false'
+          comment: 'false'
+          detailed_summary: 'true'
+          fail_on_failure: 'false'
+          fail_on_parse_error: 'true'
+          flaky_summary: 'true'
+          include_empty_in_summary: 'true'
+          include_passed: 'true'
+          include_time_in_summary: 'true'
+          report_paths: 'target/nextest/default/*junit.xml'
+          require_passed_tests: 'true'
+          require_tests: 'true'
+          simplified_summary: 'true'
+          truncate_stack_traces: 'false'
+          group_reports: 'true'
+          check_name: 'test-report-${{matrix.rust.version}}'
+          skip_success_summary: 'false'
+          job_summary: 'true'
+          verbose_summary: 'false'
 
       - id: "clippy"
         name: "run clippy"
         run: |
           just debug_justfile="${{matrix.debug_justfile}}" rust=${{matrix.rust.version}} \
             cargo clippy --all-targets --all-features -- -D warnings
 
+      - id: "docs"
+        name: "run rustdoc"
+        run: |
+          just \
+            debug_justfile="${{matrix.debug_justfile}}" \
+            rust=${{matrix.rust.version}} \
+            profile=debug \
+            target=x86_64-unknown-linux-gnu \
+            cargo doc
+          just \
+            debug_justfile="${{matrix.debug_justfile}}" \
+            rust=${{matrix.rust.version}} \
+            profile=release \
+            target=x86_64-unknown-linux-gnu \
+            cargo doc
```

Comment:
Looking at this - Do we need to run `cargo doc` with all three toolchain versions?